File: MATTHEW/leetcode/AIcoding/4.GPT-4o/go_solutions/611-valid-triangle-number.go
func triangleNumber(nums []int) int {
    sort.Ints(nums)
    count := 0
    n := len(nums)

    for i := 0; i < n-2; i++ {
        k := i + 2
        for j := i + 1; j < n-1; j++ {
            for k < n && nums[i]+nums[j] > nums[k] {
                k++
            }
            count += k - j - 1
        }
    }

    return count
}

File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/rust_solutions/611-valid-triangle-number.rs
impl Solution {
    pub fn triangle_number(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort();
        let mut count = 0;
        for i in (2..nums.len()).rev() {
            let mut left = 0;
            let mut right = i - 1;
            while left < right {
                if nums[left] + nums[right] > nums[i] {
                    count += right - left;
                    right -= 1;
                } else {
                    left += 1;
                }
            }
        }
        count as i32
    }
}